src/Stryker.Core/Stryker.Core/MutantFilters/ExcludeFromCodeCoverageFilter.cs

```diff
@@ -0,0 +1,38 @@
+using System.Collections.Generic;
+using System.Linq;
+using System.Text.RegularExpressions;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Stryker.Core.Mutants;
+using Stryker.Core.Options;
+using Stryker.Core.ProjectComponents;
+
+namespace Stryker.Core.MutantFilters
+{
+    public class ExcludeFromCodeCoverageFilter : IMutantFilter
+    {
+        string IMutantFilter.DisplayName { get; } = "exclude from code coverage filter";
+
+        IEnumerable<Mutant> IMutantFilter.FilterMutants(IEnumerable<Mutant> mutants, FileLeaf file, StrykerOptions options) =>
+            mutants.Where(m => !Exclude(m.Mutation.OriginalNode));
+
+        private static bool Exclude(SyntaxNode node) => 
+            node != null && (HasExcludeFromCodeCoverageAttribute(node) || Exclude(node.Parent));
+
+        private static bool HasExcludeFromCodeCoverageAttribute(SyntaxNode node)
+        {
+            switch (node)
+            {
+                case MemberDeclarationSyntax m:
+                    return HasExcludeFromCodeCoverageAttribute(m);
+                default:
+                    return false;
+            }
+        }
+
+        private static bool HasExcludeFromCodeCoverageAttribute(MemberDeclarationSyntax m) =>
+            m.AttributeLists
+                .SelectMany(attr => attr.Attributes)
+                .Any(attr => Regex.IsMatch(attr.ToString(), @"^(?:System\.Diagnostics\.CodeAnalysis\.)?ExcludeFromCodeCoverage(?:Attribute)?$"));
```

Comment:
I just learned that Namespace naming is enclosed in `QualifiedNameSyntax` nodes. So you could also get the `IdentifierNameSyntax` from the attribute DescendingNodesAndSelf(). We could get rid of the regex that way.

Reply:
Not sure how to fix this in the tests since I suspect it to not be available with the currently used code snippets.